hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/resolver/backend/JdbcBackendResolver.java

```diff
@@ -40,23 +45,45 @@ public JdbcBackendResolver(FQNResolver fqnResolver) {
         return Optional.empty();
       }
 
-      final String revisedBackendUriStr = backendUriStr.replaceFirst("^jdbc:", "");
-      URI backendURI = URI.create(revisedBackendUriStr);
-      String path = backendURI.getPath();
-      String dbScheme = backendURI.getScheme();
-      backendUriStr = String.format("%s:%d", backendURI.getHost(), backendURI.getPort());
+      // Split the URL based on two slashes to extract the host and port values.
+      String[] parts = backendUriStr.split("//");
+      String dbType;
+      if (parts.length > 1) {
+        backendUriStr = JDBC_EVENT_PREFIX + "://" + parts[1];
+        dbType = getDbType(parts[0]);
+      } else {
+        dbType = JDBC_EVENT_PREFIX;
+      }
+
+      String path = null;
+      try {
+        URI backendURI = URI.create(backendUriStr);
+        path = backendURI.getPath();
+        if (backendURI.getHost() != null) {
+          backendUriStr = String.format("%s:%d", backendURI.getHost(), backendURI.getPort());
+        }
+      } catch (Exception e) {
```

Comment:
I am curious about what happens in this block. Either we acquire the limiter or not, do we want to keep gonig?